pandas/core/groupby/ops.py

```diff
@@ -747,6 +747,10 @@ def group_info(self):
             ngroups,
         )
 
+    @cache_readonly
+    def recons_labels(self):
+        return [np.r_[0, np.flatnonzero(self.bins[1:] != self.bins[:-1]) + 1]]
```

Comment:
can you add some comments here on what is going on

Reply:
ok, I added it. it is inspired by here:
https://github.com/pandas-dev/pandas/blob/6cc82344fb319032465d6ee4e3d1c02991ef42c3/pandas/core/groupby/generic.py#L644-L646
that means get boundaries of ids. In my case, it would be unique result indices which is equivalent to boundaries of result indices.

I have numberlessly tested along with asv and timeit. as a result, I use bins[1:] != bins[:-1] instead of np.diff for the performance due to their dtype difference.

It was quite hard that measure the performance as it fluctuates every time in my environment. so I ran it again and again even though it takes 6~8 hours. several groups fluctuate from 0.5 ~ 1.8 but it looks like "BENCHMARKS NOT SIGNIFICANTLY CHANGED." I'm quite sure that this change won't decrease performance.